File: CS/WpfApplication1/ViewModel.cs
using System;
using System.Windows;
using System.Windows.Input;
using System.Collections.ObjectModel;
using DevExpress.Xpf.Core.Commands;

namespace WpfApplication1 {
    public class ViewModel : DependencyObject {
        public static DependencyProperty CurrentCustomerProperty =
   DependencyProperty.Register("CurrentCustomer", typeof(CustomerInfo),
                               typeof(ViewModel));
        public CustomerInfo CurrentCustomer {
            get { return (CustomerInfo)GetValue(CurrentCustomerProperty); }
            set { SetValue(CurrentCustomerProperty, value); }
        }

        ObservableCollection<CustomerInfo> customers;
        public ObservableCollection<CustomerInfo> Customers {
            get {
                if(customers == null) {
                    customers = new ObservableCollection<CustomerInfo>();
                    CustomerInfo customer = new CustomerInfo();
                    customer.Name = "Andrew Fouler";
                    customer.Salary = 13;
                    customer.Date = DateTime.Today;
                    customer.IsFired = false;
                    customers.Add(customer);

                    customer = new CustomerInfo();
                    customer.Name = "Margaret Peacock";
                    customer.Date = DateTime.Today.AddDays(1);
                    customer.Salary = 9;
                    customer.IsFired = true;
                    customers.Add(customer);

                    customer = new CustomerInfo();
                    customer.Name = "Nancy Davolio";
                    customer.Date = DateTime.Today.AddDays(2);
                    customer.Salary = 117;
                    customers.Add(customer);
                }
                return customers;
            }
        }
        public ICommand MoveNextRowCommand { get; private set; }
        public ICommand MovePrevRowCommand { get; private set; }

        public ViewModel() {
            MoveNextRowCommand = new DelegateCommand<object>(o => MoveNextRow());
            MovePrevRowCommand = new DelegateCommand<object>(o => MovePrevRow());
        }
        void MoveNextRow() {
            int currentIndex = Customers.IndexOf(CurrentCustomer);
            if(currentIndex < Customers.Count - 1)
                CurrentCustomer = Customers[currentIndex + 1];
        }
        void MovePrevRow() {
            int currentIndex = Customers.IndexOf(CurrentCustomer);
            if(currentIndex > 0)
                CurrentCustomer = Customers[currentIndex - 1];
        }
    }
}

File: VB/WpfApplication1/ViewModel.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Data
Imports System.Windows.Documents
Imports System.Windows.Input
Imports System.Windows.Media
Imports System.Windows.Media.Imaging
Imports System.Windows.Navigation
Imports System.Windows.Shapes
Imports DevExpress.Xpo
Imports System.ComponentModel
Imports DevExpress.Xpf.Grid
Imports System.Windows.Threading
Imports DevExpress.Xpf.Editors.Settings
Imports System.Threading
Imports DevExpress.Data
Imports System.Windows.Markup
Imports System.Collections.ObjectModel
Imports DevExpress.Xpf.Core.Commands

Namespace WpfApplication1
	Public Class ViewModel
		Inherits DependencyObject
		Public Shared CurrentCustomerProperty As DependencyProperty = DependencyProperty.Register("CurrentCustomer", GetType(CustomerInfo), GetType(ViewModel))
		Public Property CurrentCustomer() As CustomerInfo
			Get
				Return CType(GetValue(CurrentCustomerProperty), CustomerInfo)
			End Get
			Set(ByVal value As CustomerInfo)
				SetValue(CurrentCustomerProperty, value)
			End Set
		End Property

		Private customers_Renamed As ObservableCollection(Of CustomerInfo)
		Public ReadOnly Property Customers() As ObservableCollection(Of CustomerInfo)
			Get
				If customers_Renamed Is Nothing Then
					customers_Renamed = New ObservableCollection(Of CustomerInfo)()
					Dim customer As New CustomerInfo()
					customer.Name = "Andrew Fouler"
					customer.Salary = 13					
					customer.IsFired = False
					customers_Renamed.Add(customer)

					customer = New CustomerInfo()
					customer.Name = "Margaret Peacock"
                    customer.Salary = 9
					customer.IsFired = True
					customers_Renamed.Add(customer)

					customer = New CustomerInfo()
					customer.Name = "Nancy Davolio"
                    customer.Salary = 117
					customers_Renamed.Add(customer)
				End If
				Return customers_Renamed
			End Get
		End Property
		Private privateMoveNextRowCommand As ICommand
		Public Property MoveNextRowCommand() As ICommand
			Get
				Return privateMoveNextRowCommand
			End Get
			Private Set(ByVal value As ICommand)
				privateMoveNextRowCommand = value
			End Set
		End Property
		Private privateMovePrevRowCommand As ICommand
		Public Property MovePrevRowCommand() As ICommand
			Get
				Return privateMovePrevRowCommand
			End Get
			Private Set(ByVal value As ICommand)
				privateMovePrevRowCommand = value
			End Set
		End Property

		Public Sub New()
            MoveNextRowCommand = New DelegateCommand(Of Object)(AddressOf MoveNextRow)
            MovePrevRowCommand = New DelegateCommand(Of Object)(AddressOf MovePrevRow)
		End Sub
		Private Sub MoveNextRow()
			Dim currentIndex As Integer = Customers.IndexOf(CurrentCustomer)
			If currentIndex < Customers.Count - 1 Then
				CurrentCustomer = Customers(currentIndex + 1)
			End If
		End Sub
		Private Sub MovePrevRow()
			Dim currentIndex As Integer = Customers.IndexOf(CurrentCustomer)
			If currentIndex > 0 Then
				CurrentCustomer = Customers(currentIndex - 1)
			End If
		End Sub
	End Class
End Namespace